assets/js/state/sagas/hosts.js

```diff
@@ -1,7 +1,54 @@
-import { put, takeEvery } from 'redux-saga/effects';
-import { HOST_DEREGISTERED, removeHost } from '@state/hosts';
+import {
+  all,
+  delay,
+  put,
+  race,
+  call,
+  take,
+  takeEvery,
+} from 'redux-saga/effects';
+import {
+  DEREGISTER_HOST,
+  CANCEL_DEREGISTER_HOST,
+  HOST_DEREGISTERED,
+  removeHost,
+  setHostDeregisterable,
+} from '@state/hosts';
 import { notify } from '@state/actions/notifications';
 
+export function* markDeregisterableHosts(hosts) {
+  yield all(
+    hosts
+      .filter(({ heartbeat }) => heartbeat !== 'passing')
+      .map((host) =>
+        put({
```

Comment:
Let's put this as a function, so we don't need to copy it.
Something like:
```
export const checkHostIsDeregistarable = (host) => ({
          type: CHECK_HOST_IS_DEREGISTARABLE,
          payload: host,
})
```
In the reducer file. Until now, we have some of these in the actions files, but we agreed on not using them anymore.